### ISSUE TITLE
make fails

### ISSUE THREAD

**Issue by jrubenc** [varnishcache-friends/libvmod-geoip2#61]
Hello,

This is the same that happens at https://github.com/varnishcache-friends/libvmod-geoip2/issues/54

![image](https://github.com/varnishcache-friends/libvmod-geoip2/assets/29569449/ee421a60-7f1c-41ac-9092-503c559e9f2e)

Extra info:

$ varnishd -V
varnishd (varnish-7.0.2 revision 9b5f68e19ca0ab60010641e305fd12822f18d42c)
Copyright (c) 2006 Verdens Gang AS
Copyright (c) 2006-2021 Varnish Software

$ git status
On branch devel
Your branch is up to date with 'origin/devel'.

Steps:
$ git clone --recursive https://github.com/varnishcache-friends/libvmod-geoip2
$ cd libvmod-geoip2
$ export PKG_CONFIG_PATH=/usr/local/src/varnish-7.0.2/
$ ./autogen.sh
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal -I /usr/local/share/aclocal --force -I m4 -I ${VARNISH_DATAROOT}/aclocal
autoreconf: configure.ac: tracing
autoreconf: configure.ac: creating directory build-aux
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: copying file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal -I /usr/local/share/aclocal --force -I m4 -I ${VARNISH_DATAROOT}/aclocal
autoreconf: running: /usr/bin/autoconf --include=/usr/local/share/aclocal --force
configure.ac:5: warning: 'AM_CONFIG_HEADER': this macro is obsolete.
configure.ac:5: You should use the 'AC_CONFIG_HEADERS' macro instead.
./lib/autoconf/general.m4:2434: AC_DIAGNOSE is expanded from...
aclocal.m4:1021: AM_CONFIG_HEADER is expanded from...
configure.ac:5: the top level
configure.ac:12: warning: The macro `AC_PROG_CC_C99' is obsolete.
configure.ac:12: You should run autoupdate.
./lib/autoconf/c.m4:1659: AC_PROG_CC_C99 is expanded from...
configure.ac:12: the top level
configure.ac:13: warning: The macro `AC_PROG_LIBTOOL' is obsolete.
configure.ac:13: You should run autoupdate.
m4/libtool.m4:99: AC_PROG_LIBTOOL is expanded from...
configure.ac:13: the top level
autoreconf: running: /usr/bin/autoheader --include=/usr/local/share/aclocal --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:12: installing 'build-aux/compile'
configure.ac:13: installing 'build-aux/config.guess'
configure.ac:13: installing 'build-aux/config.sub'
configure.ac:8: installing 'build-aux/install-sh'
configure.ac:8: installing 'build-aux/missing'
src/Makefile.am: installing 'build-aux/depcomp'
parallel-tests: installing 'build-aux/test-driver'
autoreconf: Leaving directory '.'
$ ./configure

checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /usr/bin/dd
checking how to truncate binary pipes... /usr/bin/dd bs=4096 count=1
checking for mt... mt
checking if mt is a manifest tool... no
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libmaxminddb... yes
checking for a Python interpreter with version >= 3.4... python3
checking for python3... /usr/bin/python3
checking for python3 version... 3.10
checking for python3 platform... linux
checking for GNU default python3 prefix... ${prefix}
checking for GNU default python3 exec_prefix... ${exec_prefix}
checking for python3 script directory (pythondir)... ${PYTHON_PREFIX}/lib/python3.10/site-packages
checking for python3 extension module directory (pyexecdir)... ${PYTHON_EXEC_PREFIX}/lib/python3.10/site-packages
checking for rst2man... /usr/bin/rst2man
checking for vsha256.h... yes
checking for cache/cache.h... yes
checking for python3.9... /usr/bin/python3
checking for varnishtest... /usr/local/bin/varnishtest
checking for varnishd... /usr/local/sbin/varnishd
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing libtool commands
$ make
make  all-recursive
make[1]: Entering directory '/usr/local/src/1/libvmod-geoip2'
Making all in src
make[2]: Entering directory '/usr/local/src/1/libvmod-geoip2/src'
/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I..    -I/usr/local/include/varnish -I/usr/include/x86_64-linux-gnu -g -O2 -MT libvmod_geoip2_la-vmod_geoip2.lo -MD -MP -MF .deps/libvmod_geoip2_la-vmod_geoip2.Tpo -c -o libvmod_geoip2_la-vmod_geoip2.lo `test -f 'vmod_geoip2.c' || echo './'`vmod_geoip2.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I/usr/local/include/varnish -I/usr/include/x86_64-linux-gnu -g -O2 -MT libvmod_geoip2_la-vmod_geoip2.lo -MD -MP -MF .deps/libvmod_geoip2_la-vmod_geoip2.Tpo -c vmod_geoip2.c  -fPIC -DPIC -o .libs/libvmod_geoip2_la-vmod_geoip2.o
vmod_geoip2.c: In function 'vslv':
vmod_geoip2.c:60:27: error: 'NO_VXID' undeclared (first use in this function); did you mean 'OA_VXID'?
   60 |                 VSLv(tag, NO_VXID, fmt, ap);
      |                           ^~~~~~~
      |                           OA_VXID
vmod_geoip2.c:60:27: note: each undeclared identifier is reported only once for each function it appears in
vmod_geoip2.c: In function 'vmod_geoip2__init':
vmod_geoip2.c:78:24: error: 'NO_VXID' undeclared (first use in this function); did you mean 'OA_VXID'?
   78 |         VSL(SLT_Debug, NO_VXID,
      |                        ^~~~~~~
      |                        OA_VXID
make[2]: *** [Makefile:674: libvmod_geoip2_la-vmod_geoip2.lo] Error 1
make[2]: Leaving directory '/usr/local/src/1/libvmod-geoip2/src'
make[1]: *** [Makefile:486: all-recursive] Error 1
make[1]: Leaving directory '/usr/local/src/1/libvmod-geoip2'
make: *** [Makefile:397: all] Error 2

$ cat /usr/local/src/varnish-7.0.2/varnishapi.pc
prefix=/usr/local
exec_prefix=${prefix}
bindir=${exec_prefix}/bin
sbindir=${exec_prefix}/sbin
libdir=${exec_prefix}/lib
sysconfdir=${prefix}/etc
pkgsysconfdir=${sysconfdir}/varnish
includedir=${prefix}/include
pkgincludedir=${includedir}/varnish
datarootdir=${prefix}/share
datadir=${datarootdir}
pkgdatadir=${datadir}/varnish
vcldir=${pkgdatadir}/vcl
vmoddir=${libdir}/varnish/vmods
vmodtool=${pkgdatadir}/vmodtool.py
vsctool=${pkgdatadir}/vsctool.py

Name: VarnishAPI
Description: Varnish API
Version: 7.0.2
Cflags: -I${pkgincludedir}
Libs: -L${libdir} -lvarnishapi

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.3 LTS
Release:        22.04
Codename:       jammy

$ gcc --version
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0

**Comment by jrubenc** [comment]
But then:

$ make
make  all-recursive
make[1]: Entering directory '/usr/local/src/1/libvmod-geoip2'
Making all in src
make[2]: Entering directory '/usr/local/src/1/libvmod-geoip2/src'
/usr/bin/python3  ../src/vmod_geoip2.vcc
  File "/usr/local/src/1/libvmod-geoip2/src/../src/vmod_geoip2.vcc", line 28
    $Module geoip2 3 "Varnish GeoIP2 Lookup Module"
    ^
SyntaxError: invalid syntax
make[2]: *** [Makefile:1122: vcc_if.c] Error 1
make[2]: Leaving directory '/usr/local/src/1/libvmod-geoip2/src'
make[1]: *** [Makefile:486: all-recursive] Error 1
make[1]: Leaving directory '/usr/local/src/1/libvmod-geoip2'
make: *** [Makefile:397: all] Error 2

**Comment by fgsch** [comment]
I'm going to close this for now. If it's still happening please comment here or open a new issue. Thank you!
